app/views/courses/show.html.erb

```diff
@@ -6,15 +6,13 @@
           <%= render 'shared/components/icon_button_primary', icon_name: 'icon-download' %>
         <% end %>
       <% end %>
-      <% if policy(@course).destroy? %>
-        <%= link_to alert_modal_path(
+      <%= link_to alert_modal_path(
               title: t("alert.title", resource_name: "course"),
               description: t("alert.description", label: @course.title, resource_name: "course"),
               action_path: course_path(@course),
               method: :delete,
-            ), data: { turbo_frame: "modal" } do %>
-          <%= render 'shared/components/outline_button_danger', label: 'Delete' %>
-        <% end %>
+            ), class: @course.has_enrollments? ? 'disabled' : '', data: { turbo_frame: "modal" } do %>
```

Comment:
we need to check if the course is published too. Published courses can't be deleted. The policy will break and an exception will be raised here.

Reply:
Better to add a permission_helper.rb and add can_delete? method for the given resource.